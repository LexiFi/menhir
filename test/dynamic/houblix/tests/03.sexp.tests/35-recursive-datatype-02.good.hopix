type t = N | P (t, t)

let leaf : t = N<>

let a_tree : t =
  P<> (P<> (P<> (leaf, leaf), P<>(leaf, P<>(leaf, leaf))), leaf)