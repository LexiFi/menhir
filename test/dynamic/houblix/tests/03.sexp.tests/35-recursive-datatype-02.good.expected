(((value(DefineType((value(TCon t))(position((start_p(""1 0 5))(end_p(""1 0 6)))))()(DefineSumType((((value(KId N))(position((start_p(""1 0 9))(end_p(""1 0 10)))))())(((value(KId P))(position((start_p(""1 0 13))(end_p(""1 0 14)))))(((value(TyCon(TCon t)()))(position((start_p(""1 0 16))(end_p(""1 0 17)))))((value(TyCon(TCon t)()))(position((start_p(""1 0 19))(end_p(""1 0 20)))))))))))(position((start_p(""1 0 0))(end_p(""1 0 21)))))((value(DefineValue(SimpleValue(((value(Id leaf))(position((start_p(""3 23 27))(end_p(""3 23 31)))))(((value(ForallTy()((value(TyCon(TCon t)()))(position((start_p(""3 23 34))(end_p(""3 23 35)))))))(position((start_p(""3 23 33))(end_p(""3 23 35))))))((value(Tagged((value(KId N))(position((start_p(""3 23 38))(end_p(""3 23 39)))))(())()))(position((start_p(""3 23 38))(end_p(""3 23 41)))))))))(position((start_p(""3 23 23))(end_p(""3 23 41)))))((value(DefineValue(SimpleValue(((value(Id a_tree))(position((start_p(""5 43 47))(end_p(""5 43 53)))))(((value(ForallTy()((value(TyCon(TCon t)()))(position((start_p(""5 43 56))(end_p(""5 43 57)))))))(position((start_p(""5 43 55))(end_p(""5 43 57))))))((value(Tagged((value(KId P))(position((start_p(""6 60 62))(end_p(""6 60 63)))))(())(((value(Tagged((value(KId P))(position((start_p(""6 60 67))(end_p(""6 60 68)))))(())(((value(Tagged((value(KId P))(position((start_p(""6 60 72))(end_p(""6 60 73)))))(())(((value(Variable((value(Id leaf))(position((start_p(""6 60 77))(end_p(""6 60 81)))))()))(position((start_p(""6 60 77))(end_p(""6 60 81)))))((value(Variable((value(Id leaf))(position((start_p(""6 60 83))(end_p(""6 60 87)))))()))(position((start_p(""6 60 83))(end_p(""6 60 87))))))))(position((start_p(""6 60 72))(end_p(""6 60 88)))))((value(Tagged((value(KId P))(position((start_p(""6 60 90))(end_p(""6 60 91)))))(())(((value(Variable((value(Id leaf))(position((start_p(""6 60 94))(end_p(""6 60 98)))))()))(position((start_p(""6 60 94))(end_p(""6 60 98)))))((value(Tagged((value(KId P))(position((start_p(""6 60 100))(end_p(""6 60 101)))))(())(((value(Variable((value(Id leaf))(position((start_p(""6 60 104))(end_p(""6 60 108)))))()))(position((start_p(""6 60 104))(end_p(""6 60 108)))))((value(Variable((value(Id leaf))(position((start_p(""6 60 110))(end_p(""6 60 114)))))()))(position((start_p(""6 60 110))(end_p(""6 60 114))))))))(position((start_p(""6 60 100))(end_p(""6 60 115))))))))(position((start_p(""6 60 90))(end_p(""6 60 116))))))))(position((start_p(""6 60 67))(end_p(""6 60 117)))))((value(Variable((value(Id leaf))(position((start_p(""6 60 119))(end_p(""6 60 123)))))()))(position((start_p(""6 60 119))(end_p(""6 60 123))))))))(position((start_p(""6 60 62))(end_p(""6 60 124)))))))))(position((start_p(""5 43 43))(end_p(""6 60 124))))))
